pinot-core/src/main/java/com/linkedin/pinot/core/startree/StarTreeDataTable.java

```diff
@@ -211,9 +192,9 @@ public boolean hasNext() {
       public Pair<byte[], byte[]> next() {
         byte[] dimensionBytes = new byte[_dimensionSize];
         byte[] metricBytes = new byte[_metricSize];
-        ByteBuffer byteBuffer = _dataBuffer.toDirectByteBuffer(_currentIndex++ * _docSize, (int) _docSize);
-        byteBuffer.get(dimensionBytes);
-        byteBuffer.get(metricBytes);
+        ByteBuffer byteBuffer = _dataBuffer.toDirectByteBuffer(_currentIndex++ * _docSizeLong, (int) _docSizeLong);
```

Comment:
Seems we changed _docSize name to _docSizeLong, and then casting it here to int. I am sure the logic is correct, but makes this part of the code a bit confusing.

Reply:
Fixed